becquerel/core/spectrum.py

```diff
@@ -973,12 +973,15 @@ def find_bin_index(self, x, use_kev=True):
 
         bin_edges = self.bin_edges_kev if use_kev else self.bin_edges_raw
         bin_widths = self.bin_widths_kev if use_kev else self.bin_widths_raw
+        x = np.array(x)
 
-        assert x >= bin_edges[0]
-        assert x < bin_edges[-1]
+        if np.any(x < bin_edges[0]):
+            raise ValueError('requested x is < lowest bin edge')
+        if np.any(x >= bin_edges[-1]):
+            raise ValueError('requested x is >= highest bin edge')
 
         if self.has_uniform_bins(use_kev=use_kev):
-            return int((x - bin_edges[0]) / bin_widths[0])
+            return ((x - bin_edges[0]) / bin_widths[0]).astype(np.int)
```

Comment:
this changes the output type for scalar input. I think this should return scalar output for scalar input (as the docstring implies). See:

https://docs.scipy.org/doc/numpy/reference/generated/numpy.isscalar.html

You could not do the conversion (the edge checks still work) and then handle the dtype based on `isscalar`

Reply:
As we found offline, this (inadvertently) takes advantage of some subtle [numpy casting rules](https://docs.scipy.org/doc/numpy/reference/generated/numpy.can_cast.html). With the `x = np.array(x)` above these lines, a scalar input `x` will still return an `np.int` rather than an array. This is why the tests passed and the notebooks looked correct. I may add some more type-checking to the tests to be safe.

Reply:
Yeah a zero dim array with int dtype behaves like an int. Good stuff m!